File: Task_Angular_v19/src/app/modules/admin/services/admin.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { StorageService } from '../../../auth/services/storage/storage.service';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})

export class AdminService {

  constructor(private http: HttpClient) { }

  getUsers(): Observable<any> {
    return this.http.get('http://localhost:8080/api/admin/users', httpOptions);
  }

  postTask(taskDto: any): Observable<any> {
    return this.http.post('http://localhost:8080/api/admin/task', taskDto, httpOptions);
  }

  searchTasks(title: string): Observable<any> {
    return this.http.get('http://localhost:8080/api/admin/tasks/search/' + title, httpOptions);
  }

  deleteTask(id: number): Observable<any> {
    return this.http.delete('http://localhost:8080/api/admin/task/' + id, httpOptions);
  }

  updateTask(id: number, taskDTO: any): Observable<any> {
    return this.http.put('http://localhost:8080/api/admin/task/' + id, taskDTO, httpOptions);
  }

  getTaskById(id: number): Observable<any> {
    return this.http.get('http://localhost:8080/api/admin/task/' + id, httpOptions);
  }

  getCommentsById(id: number): Observable<any> {
    return this.http.get('http://localhost:8080/api/admin/task/' + id + '/comments', httpOptions);
  }

  createComment(taskId: number, content: string): Observable<any> {
    const params_ = {
      taskId: taskId,
      postedBy:StorageService.getUserId() ?? ''
    };
    return this.http.post('http://localhost:8080/api/admin/task/comment', content, {
      params: params_,
      headers: new HttpHeaders({
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${StorageService.getToken()}`
      }),
      withCredentials: true
    });
  }


  getTasks(): Observable<any> {
    console.log('Current Token:', StorageService.getToken());
    console.log('httpOptions:', httpOptions);
    return this.http.get('http://localhost:8080/api/admin/tasks', httpOptions);
  }


}
const httpOptions = {
  headers: new HttpHeaders({
    'Content-Type': 'application/json',
    'Authorization': `Bearer ${StorageService.getToken()}`
  }),
  withCredentials: true
};

File: TaskManagement/src/main/java/com/finalproject/TaskManagement/services/admin/AdminService.java
package com.finalproject.TaskManagement.services.admin;

import com.finalproject.TaskManagement.dtos.CommentDTO;
import com.finalproject.TaskManagement.dtos.TaskDTO;
import com.finalproject.TaskManagement.dtos.UserDTO;

import java.util.List;

/**
 * Интерфейс для сервиса управления администрацией.
 * Определяет методы для работы с пользователями, задачами и комментариями.
 */
public interface AdminService {

    /**
     * Получает список всех пользователей.
     *
     * @return список DTO объектов пользователей
     */
    List<UserDTO> getUsers();

    /**
     * Создает новую задачу.
     *
     * @param taskDTO DTO объекта задачи
     * @return созданная задача в виде DTO
     */
    TaskDTO postTask(TaskDTO taskDTO);

    /**
     * Получает список всех задач.
     *
     * @return список DTO объектов задач
     */
    List<TaskDTO> getTasks();

    /**
     * Получает задачу по идентификатору.
     *
     * @param id идентификатор задачи
     * @return задача в виде DTO
     */
    TaskDTO getTaskById(Long id);

    /**
     * Удаляет задачу по идентификатору.
     *
     * @param id идентификатор задачи
     */
    void deleteTaskById(Long id);

    /**
     * Ищет задачи по заголовку.
     *
     * @param title подстрока для поиска
     * @return список задач, соответствующих запросу
     */
    List<TaskDTO> searchTaskByTitle(String title);

    /**
     * Обновляет задачу по идентификатору.
     *
     * @param id идентификатор задачи
     * @param taskDTO объект DTO с обновленными данными
     * @return обновленная задача в виде DTO
     */
    TaskDTO updateTask(Long id, TaskDTO taskDTO);

    /**
     * Создает комментарий для задачи.
     *
     * @param taskId  идентификатор задачи
     * @param postedBy идентификатор пользователя, опубликовавшего комментарий
     * @param content содержимое комментария
     * @return созданный комментарий в виде DTO
     */
    CommentDTO createComment(Long taskId, Long postedBy, String content);

    /**
     * Получает список комментариев по идентификатору задачи.
     *
     * @param taskId идентификатор задачи
     * @return список комментариев в виде DTO
     */
    List<CommentDTO> getCommentsByTask(Long taskId);
}
